powersimdata/data_access/data_access.py

```diff
@@ -27,6 +30,29 @@ def __init__(self, root):
         self.root = root
         self.join = fs2.path.join
 
+    def read(self, filepath):
+        """Reads data from local machine.
+
+        :param str filepath: path to file, with extension either 'pkl', 'csv', or 'mat'.
+        :return: (*pandas.DataFrame* or *dict*) -- pkl and csv files will be returned as
+            a data frame, while a mat file will be returned as a dictionary
+        :raises ValueError: if extension is unknown.
+        """
+        ext = os.path.basename(filepath).split(".")[-1]
+        if ext == "pkl":
+            file_object = self.fs.open(filepath, mode="rb")
+            data = pd.read_pickle(file_object)
+        elif ext == "csv":
+            file_object = self.fs.open(filepath)
```

Comment:
Any reason we don't use `mode="rb"` here as well?

Reply:
To be safe, I would close the `file_object` when we're done with it. This could be just before the return, or if we can standardize the mode, we could move the call outside the if/else and use a `with` statement.

Reply:
Good call! I had to use `rb` mode for the `mat` file, and didn't realize it would work for `csv` as well. Updated.